lib/adapters/show-document-adapter.ts

```diff
@@ -0,0 +1,43 @@
+import * as url from 'url';
+import { shell } from 'electron';
+import {
+  LanguageClientConnection,
+  ShowDocumentParams,
+  ShowDocumentResult,
+} from '../languageclient';
+import Convert from '../convert';
+
+/** Public: Adapts the window/showDocument command to Atom's text editors or external programs. */
+export default class ShowDocumentAdapter {
+  /**
+   * Public: Attach to a {LanguageClientConnection} to recieve requests to show documents.
+   */
+  public static attach(connection: LanguageClientConnection): void {
+    connection.onShowDocument(ShowDocumentAdapter.onShowDocument);
+  }
+
+  /**
+   * Public: Show a notification message with buttons using the Atom notifications API.
+   *
+   * @param params The {ShowDocumentParams} received from the language server
+   *   indicating the document to be displayed as well as other metadata.
+   */
+  public static async onShowDocument(params: ShowDocumentParams): Promise<ShowDocumentResult> {
+    if (!params.external) {
+      const editor = await atom.workspace.open(params.uri, {
+        activateItem: params.takeFocus,
+        activatePane: params.takeFocus,
+        pending: true,
+        initialLine: params.selection?.start.line ?? 0,
+        initialColumn: params.selection?.start.character ?? 0,
+      });
+      if (params.selection != null) {
+        editor.selectToBufferPosition(Convert.positionToPoint(params.selection.end));
```

Comment:
@aminya TypeScript trips up here because the DefinitelyTyped types don't mention that `atom.workspace.open` returns a `Promise<TextEditor>`. I've submitted a PR to that repo, you should be mentioned to review it. That will fix the other CI error

Reply:
Ongoing PR:
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/50382

Reply:
Alright, worked around it.